sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallAsyncClient.java

```diff
@@ -0,0 +1,513 @@
+                    .createCallWithResponseAsync(request, contextValue)
+                    .flatMap(response -> Mono.just(new SimpleResponse<>(response, response.getValue())));
+            });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Play audio in a call.
+     *
+     * @param callId The call id.
+     * @param audioFileUri The media resource uri of the play audio request.
+     * @param loop The flag indicating whether audio file needs to be played in loop or not.
+     * @param audioFileId An id for the media in the AudioFileUri, using which we cache the media.
+     * @param callbackUri call back uri to receive notifications.
+     * @param operationContext The value to identify context of the operation.
+     * @return the response payload for play audio operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PlayAudioResponse> playAudio(String callId,
+                                             String audioFileUri,
+                                             boolean loop,
+                                             String audioFileId,
+                                             String callbackUri,
+                                             String operationContext) {
+        try {
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
+            Objects.requireNonNull(audioFileUri, "'audioFileUri' cannot be null.");
+            PlayAudioRequest playAudioRequest =
+                PlayAudioConverter.convert(audioFileUri, loop, audioFileId, callbackUri, operationContext);
+            return playAudio(callId, playAudioRequest);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Play audio in a call.
+     *
+     * @param callId The call id.
+     * @param audioFileUri The media resource uri of the play audio request.
+     * @param playAudioOptions Options for play audio.
+     * @return the response payload for play audio operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PlayAudioResponse> playAudio(String callId,
+                                             String audioFileUri,
+                                             PlayAudioOptions playAudioOptions) {
+        try {
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
+            Objects.requireNonNull(audioFileUri, "'audioFileUri' cannot be null.");
+            PlayAudioRequest playAudioRequest = PlayAudioConverter.convert(audioFileUri, playAudioOptions);
+            return playAudio(callId, playAudioRequest);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<PlayAudioResponse> playAudio(String callId, PlayAudioRequest playAudioRequest) {
+        try {
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
```

Comment:
You have already checked callId is not null in line 134 and 157. Let's remove the duplication and just keep the one at 168.

Reply:
duplicate checks removed